src/resolve/resolved-mdns.c

```diff
@@ -397,6 +406,20 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
                                 log_debug("Got a goodbye packet");
                                 /* See the section 10.1 of RFC6762 */
                                 rr->ttl = 1;
+
+                                /* Look at the cache 1 second later and remove stale entries.
+                                 * This is particularly useful to keep service browsers updated on service removal,
+                                 * as there are no other reliable triggers to propogate that info. */
+                                r = sd_event_add_time_relative(
```

Comment:
so this will create one event source for each incoming mdns packet with ttl set to 0. that is a chance for a DoS, as clients can make us allocate basically unbounded memory, if they send us a ton of these packets within 1s.

Thus the event source object should be stored somewhere (in the scope object?), and if it already exists no new event source should be allocated. morevoer, the handler should check if there are RRs in the cache that should be flushed within less than 1s, and if so should immediately requeue a new event source to cover for that.

That way there's only one event source enqueued at max.